### PR TITLE
Stationary solver and LpSmoother Operator impls

### PR DESCRIPTION
**Pull request by aujxn**
`StationaryIteration` with `LpSmoother` for preconditioner is simple and parallel smoother for AMG

### CODE REVIEW

File: algebra/sfem_LpSmoother.hpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        std::shared_ptr<Buffer<T>> inv_diag;
+
+        void default_init() {
+            OpenMP_BLAS<T>::build_blas(blas);
+            execution_space_ = EXECUTION_SPACE_HOST;
+        }
+
+        /* Operator */
+        int apply(const T* const b, T* const x) override {
+            blas.axpyz(n_dofs, inv_diag, b, 1, x);
```
Comment (aujxn) [inline algebra/sfem_LpSmoother.hpp:27]: Ah I guess I should have used alpha = 0 here

Comment (zulianp) [inline algebra/sfem_LpSmoother.hpp:27]: This should maybe be `xypaz` ?

Comment (zulianp) [reply algebra/sfem_LpSmoother.hpp:27]: Actually I always allow Operators to add, so this is fine

Comment (zulianp) [reply algebra/sfem_LpSmoother.hpp:27]: In fact I set the output vector to zero if I want to make sure there is no add outside

File: algebra/sfem_Stationary.hpp [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        void set_op(const std::shared_ptr<Operator<T>>& op) override {
+            this->apply_op = [=](const T* const x, T* const y) { op->apply(x, y); };
+            n_dofs = op->rows();
+        }
+        void set_preconditioner_op(const std::shared_ptr<Operator<T>>& op) override {
+            this->preconditioner_op = [=](const T* const x, T* const y) { op->apply(x, y); };
+        }
+        void set_max_it(const int it) override { max_it = it; }
+        void set_n_dofs(const ptrdiff_t n) override { this->n_dofs = n; }
+        void set_initial_guess_zero(const bool val) override { is_initial_guess_zero = val; }
```
Comment (aujxn) [inline algebra/sfem_Stationary.hpp:58]: And maybe assert(false) here since initial guess is always going to be x

Comment (zulianp) [reply algebra/sfem_Stationary.hpp:58]: You can just ignore callback and remove `is_initial_guess_zero` field. This is just more information coming from outside allowing for solver specific (small) optimisations

File: algebra/sfem_LpSmoother.hpp (hunk 1 above)
Comment (aujxn) [reply algebra/sfem_LpSmoother.hpp]: but it does feel strange to expect the preconditioner to apply the correction so maybe this inconsistency is not good. That's how I left it in the new commit though

Comment (aujxn) [reply algebra/sfem_LpSmoother.hpp]: yep

**Pull request merged**